Sources/NIO/Linux.swift

```diff
@@ -132,5 +135,54 @@ internal enum Linux {
         }
         return fd
     }
+
+    private static func contentsOfFile(path: String) throws -> Substring {
+        let fh = try NIOFileHandle(path: path)
+        defer { try! fh.close() }
+        // linux doesn't properly report /sys/fs/cgroup/* files lengths so we use a reasonable limit
+        var buf = ByteBufferAllocator().buffer(capacity: 1024)
+        try buf.writeWithUnsafeMutableBytes(minimumWritableBytes: buf.capacity) { ptr in
+            let res = try fh.withUnsafeFileDescriptor { fd -> IOResult<ssize_t> in
+                return try Posix.read(descriptor: fd, pointer: ptr.baseAddress!, size: ptr.count)
+            }
+            switch res {
+            case .processed(let n):
+                return n
+            case .wouldBlock:
+                preconditionFailure("read returned EWOULDBLOCK despite a blocking fd")
+            }
+        }
+        return String(buffer: buf).prefix(while: { $0 != "\n" })
```

Comment:
Maybe this function should only return `String` instead of `Substring` and this `prefix(while:)` should move elsewhere? Otherwise it's a bit weird that this is called `contentsOfFile`, when it's really `firstLineOfFile`.

Reply:
I renamed it to `firstLineOfFile` to be more correct. :)